src/main/java/core/mate/academy/service/MachineService.java

```diff
@@ -1,11 +1,12 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
 import java.util.List;
 
 /**
  * Parameterize this service and add its implementation in a separate class.
  */
-public interface MachineService {
+public interface MachineService<M> {
```

Comment:
```suggestion
public interface MachineService<T> {
```
better to call generics `T`. See naming convention: https://docs.oracle.com/javase/tutorial/java/generics/types.html